### PR TITLE
Release fix/37

### LINKED ISSUE #37: Feature 17 conversion of day enum from api-client
**Describe the bug**
PostActivity from feature 17 does not convert Days correctly.

**To Reproduce**
1. Use the web-api PostActivity endpoint through the flutter application.
2. Conversion from flutter applications Weekday enum to web-api Days does not work

**Expected behavior**
Conversion should work correctly. i.e friday goes to friday

**Actual behavior**
Conversion does not work correctly. i.e friday goes to index 0 (nothing)

**Environment (please complete the following information):**
MariaDB web-client
Windows 10
Android emulator pixel_c

### PR DESCRIPTION
**Pull request by KVISDAOWNER**
Fixes #37

### CODE REVIEW

**Comment by KVISDAOWNER** [comment]
@aau-giraf/sw613f19  you have been assigned this. Here's your checklist. I marked @HaZzM22  as the reviewer, but you might change it as you like.
Please use the following checklist. Feel free to check it off as you go along.
**Code Design**
- [ ] Is the code in the right place? (Both in terms of folder structure and class structure)
- [ ] Could this code, to the best of your knowledge, have reused existing code?
- [ ] Is the code over-engineered? Examples of over-engineering: Implemented behavior for future problems
- [ ] Does it introduce functionality that is not necessary for solving the problem?

**Code Readability and Maintainability**
- [ ] Are names meaningful and self-documenting?
- [ ] Is it understandable, by reading the code, what it does?
- [ ] Is the code simple and readable? i.e. it contains no “hack”, or obscure solution.
- [ ] Is it understandable, by reading the tests, what they do?
- [ ] Do the tests cover sensible cases? Both happy paths and exception paths?
- [ ] Do the tests cover the full functionality? i.e., will the tests fail if some of the requested functionality is missing?
- [ ] Are error messages in the code understandable?
- [ ] Are confusing sections of code covered by comments?

**Functionality**
For this part, open the app, and test:
- [ ] Are the features that the code claims to implement, actually implemented?
- [ ] Does it look like the code has bugs?
- [ ] Are all new screens reachable through in-app navigation?

**Points for Discussion and Reflection**
- [ ] Are UI messages, that the user might meet, actually helpful?
- [ ] Are there errors that are bad practice? Like, hard-coding a test database connection?
- [ ] Is there well written code in this? Remember to tell the developer!

**Comment by KVISDAOWNER** [comment]
@aau-giraf/sw609f19  you have been assigned this. Here's your checklist. I marked @Tricky12321  as the reviewer, but you might change it as you like.
Please use the following checklist. Feel free to check it off as you go along.
**Code Design**
- [ ] Is the code in the right place? (Both in terms of folder structure and class structure)
- [ ] Could this code, to the best of your knowledge, have reused existing code?
- [ ] Is the code over-engineered? Examples of over-engineering: Implemented behavior for future problems
- [ ] Does it introduce functionality that is not necessary for solving the problem?

**Code Readability and Maintainability**
- [ ] Are names meaningful and self-documenting?
- [ ] Is it understandable, by reading the code, what it does?
- [ ] Is the code simple and readable? i.e. it contains no “hack”, or obscure solution.
- [ ] Is it understandable, by reading the tests, what they do?
- [ ] Do the tests cover sensible cases? Both happy paths and exception paths?
- [ ] Do the tests cover the full functionality? i.e., will the tests fail if some of the requested functionality is missing?
- [ ] Are error messages in the code understandable?
- [ ] Are confusing sections of code covered by comments?

**Functionality**
For this part, open the app, and test:
- [ ] Are the features that the code claims to implement, actually implemented?
- [ ] Does it look like the code has bugs?
- [ ] Are all new screens reachable through in-app navigation?

**Points for Discussion and Reflection**
- [ ] Are UI messages, that the user might meet, actually helpful?
- [ ] Are there errors that are bad practice? Like, hard-coding a test database connection?
- [ ] Is there well written code in this? Remember to tell the developer!

File: GirafRest/Controllers/ActivityController.cs [hunk 1]
```diff
@@ -37,13 +37,14 @@ public ActivityController(IGirafService giraf, ILoggerFactory loggerFactory, IAu
         /// <param name="weekplanName">name of the weekplan that you want to add the activity on.</param>
         /// <param name="weekYear">year of the weekplan that you want to add the activity on.</param>
         /// <param name="weekNumber">week number of the weekplan that you want to add the activity on.</param>
-        /// <param name="weekDay">day of the week that you want to add the activity on.</param>
+        /// <param name="weekDayNmb">day of the week that you want to add the activity on (Monday=1, Sunday=7).</param>
         /// <returns>Returns <see cref="ActivityDTO"/> for the requested activity on success else MissingProperties, 
         /// UserNotFound, NotAuthorized, WeekNotFound or InvalidDay.</returns>
-        [HttpPost("{userId}/{weekplanName}/{weekYear}/{weekNumber}/{weekDay}")]
+        [HttpPost("{userId}/{weekplanName}/{weekYear}/{weekNumber}/{weekDayNmb}")]
```
Comment (Tricky12321) [inline GirafRest/Controllers/ActivityController.cs]: Has the endpoint changed?

**Comment by HaZzM22** [comment]
Seems to be working fine with ints. However when using the new endpoint with enums from the weekplanner, they should be decreased by a value of 1.

expected behavior Friday is Friday
actual behavior Friday is Thursday.

**Comment by KVISDAOWNER** [comment]
> Seems to be working fine with ints. However when using the new endpoint with enums from the weekplanner, they should be decreased by a value of 1.
> 
> expected behavior Friday is Friday
> actual behavior Friday is Thursday.

Yes, there is an issue with the enum conversion but I believe the day index should be incremented from the api client

File: GirafRest/Controllers/ActivityController.cs (hunk 1 above)
Comment (KVISDAOWNER) [reply GirafRest/Controllers/ActivityController.cs]: If I understand your question correctly: The endpoint has changed from taking a enum Day parameter named weekDay to indstead taking an int parameter named weekDayNmb, and also typecasting the int to a enum Day is added

Comment (Tricky12321) [reply GirafRest/Controllers/ActivityController.cs]: Okay, so the URL used for the endpoint has not changed?

Comment (KVISDAOWNER) [reply GirafRest/Controllers/ActivityController.cs]: No, only the parameter name and type.
Before the url could be 
"..v2/ActivityController/1/weekplan/2019/1/Monday"
now the corresponding url will be
"":..v2/ActivityController/1/weekplan/2019/1/1"

**Pull request merged**